### PR TITLE
Calendar model

### LINKED ISSUE #39: Feature : when no api version is provided use the most recent one
I think it's good to have this feature so that when we publicize the API, it can be easily accessed by front-end developers and other third-party apps.

### LINKED ISSUE #77: Creating a calendar Model
In this issue I want to discuss how should we create  the calendar model 
I have create some sort of diagram just to brain storm some ideas about the calendar and its functionalities .

__
image

this is just a basic example to be modified later,
please leave your opinions on this.

### PR DESCRIPTION
**Pull request by bellaabdelouahab**
here is a summary of the tasks made in this pull request,

- [x] Update unit tests
- [x] Fix Jest command not found issue
- [x] Add new implementation of API version
- [x] Configure Jest setup and teardown
- [x] Add calendar participant functionality
- [x] Add front end again (#98)
- [x] Fix access boolean in environment
- [x] Create Jest configuration file
- [x] Fix Swagger loading after app starts
- [x] Fix API auto versioning and clean up code
- [x] Feature: Add bearer token extractor
- [x] Fix: Add .gitignore in base directory
- [x] Fix: Update calendar routes
- [x] Fix: Add metadata to event model
- [x] Enhance: Add base routes for Calendar 

looking forward to for your reviews 

@yassineboujrada @Khalid1G @muttaqin1 

- Resolves #39 
- Resolves #77

### CODE REVIEW

File: backend-app/tests/e2e/auth/login_signup.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+describe('POST /api/auth/login', () => {
+    // console.log('db connected: ', mongoose.connection.readyState);
+
+    beforeAll(async () => {
+        await createUser(
+            app,
+            testUserCredentials.userName,
+            testUserCredentials.userEmail,
+            testUserCredentials.userPassword
+        );
```
Comment (muttaqin1) [inline backend-app/tests/e2e/auth/login_signup.test.js:16]: We can use database seeding. What do you think?

**Review by muttaqin1 — APPROVED** [review]
Great job!

File: backend-app/tests/e2e/auth/login_signup.test.js (hunk 1 above)
Comment (bellaabdelouahab) [reply backend-app/tests/e2e/auth/login_signup.test.js:16]: yes it is a great idea i already created seedData.js, you can start with that and create an issue for it so we can work on it 
@muttaqin1

File: backend-app/controllers/calendar_controllers/calendar_base_controller.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        );
+    }
+    // TODO: update calendar
+};
+
+exports.deleteCalendar = async (req, res, next) => {
+    const calendar = await calendar_validators.validateCalendar(req);
+    // check if user is not admin nor the owner of the calendar
+    if (
+        calendar.createdBy !== req.user.id ||
```
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/controllers/calendar_controllers/calendar_base_controller.js:23]: done

File: backend-app/controllers/calendar_controllers/calendar_base_controller.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const AppError = require('../utils/app_error');
+const calendar_validators = require('./calendar_validators');
+
+exports.updateCalendar = async (req, res, next) => {
+    const calendar = await calendar_validators.validateCalendar(req);
+    // check if user is not admin nor the owner of the calendar
+    if (
+        calendar.createdBy !== req.user.id ||
```
[the remark this answers is not in the document]
Comment (bellaabdelouahab) [reply backend-app/controllers/calendar_controllers/calendar_base_controller.js:8]: done

**Comment by bellaabdelouahab** [comment]
tasks 

- [ ] restructure controllers folder
- [ ] fix user id parsing
- [ ] user project version as api version
@muttaqin1

**Pull request merged**